app/Repositories/H5pLibrary/H5pLibraryRepository.php

```diff
@@ -0,0 +1,184 @@
+{
+    /**
+     * H5pLibraryRepository constructor.
+     *
+     * @param H5pLibrary $model
+     */
+    public function __construct(H5pLibrary $model)
+    {
+        parent::__construct($model);
+    }
+
+    /**
+     * Get the libraries's fields semantics.
+     *
+     * @param Object $h5pLibraryObject
+     * @return array
+     */
+    public function getFieldSemantics($h5pLibraryObject)
+    {
+        if ($h5pLibraryObject->semantics) {
+            $semantics = json_decode($h5pLibraryObject->semantics, true);
+            $results = array();
+            $fieldTypes = ['text'];
+            $this->getNestedFieldSemantics($h5pLibraryObject->id, $semantics, 'type', $results, [], [], $fieldTypes);
+            return $results;
+        }
+
+        return [];
+    }
+
+    /**
+     * Get the Nested Fields Semantics By Recursion
+     *
+     * @param  int      $libraryId
+     * @param  array    $array
+     * @param  string   $key
+     * @param  array    $results
+     * @param  array    $pathName
+     * @param  array    $pathType
+     * @param  array    $values
+     * @param  array    $nameLike
+     * @return array
+     */
+    private function getNestedFieldSemantics($libraryId, $array, $key, &$results, $pathName, $pathType, $values = null, $nameLike = null)
+    {
+        if (!is_array($array)) {
+            return;
+        }
+
+        if (isset($array['name']) && isset($array['type'])) {
+            $pathName[] = $array['name'];
+            $pathType[] = $array['type'];
+        }
+
+        if (isset($array[$key]) && $array[$key] == 'library') {
+            $pathName[] = 'params';
+            $pathType[] = $array['type'];
+
+            foreach ($array['options'] as $libraryName) {
+                $libraryNameArray = explode(' ', $libraryName);
+                $libraryVersions = explode('.', $libraryNameArray[1]);
+                $libraryObj = $this->model->firstWhere([
+                    ['name', '=', $libraryNameArray[0]],
+                    ['major_version', '=', $libraryVersions[0]],
+                    ['minor_version', '=', $libraryVersions[1]]
+                ]);
+
+                if ($libraryObj && $libraryObj->semantics) {
+                    $librarySemantics = json_decode($libraryObj->semantics, true);
+                }
+
+                $this->getNestedFieldSemantics($librarySemantics, $key, $results, $pathName, $pathType, $values, $nameLike);
+            }
+        }
+
+        if (isset($array[$key]) && $values && in_array($array[$key], $values, true)) {
+            $fields = array();
+            $fields['indexed'] = false;
+            $fields['label'] = '';
+            $fields['description'] = '';
+
+            if ($array[$key] == 'text') {
+                $pathTypeCount = count($pathType);
+
+                if ($pathTypeCount > 2 && $pathType[$pathTypeCount - 3] == 'list' && $pathType[$pathTypeCount - 2] == 'group'){
```

Comment:
Space before the opening brace.
Also, use === instead of == where we're doing string comparison.
Do the above for all instances that you find.

Reply:
Added space before the opening brace.
Used === instead of == where we're doing string comparison.
Did the above for all instances that i could find.